parsing/location.ml

```diff
@@ -174,6 +174,23 @@ let absolute_path s = (* This function could go into Filename *)
   in
   aux s
 
+let absolute_path_always s = (* This function could go into Filename *)
+  let open Filename in
+  let s =
+    if not (is_relative s) then (rewrite_absolute_path s)
+    else (rewrite_absolute_path (concat (Sys.getcwd ()) s))
+  in
+  (* Now simplify . and .. components *)
+  let rec aux s =
+    let base = basename s in
+    let dir = dirname s in
+    if dir = s then dir
+    else if base = current_dir_name then aux dir
+    else if base = parent_dir_name then dirname (aux dir)
+    else concat (aux dir) base
+  in
+  aux s
```

Comment:
I don't like the amount of code duplication with the function `absolute_path` above, and I am not fond of the name either -- `_always` suggests that we always do something, but what?

If we need to have this feature of deciding locally to rewrite only relative paths or also absolute paths, I would propose the following interface:

```
val absolute_path : rewrite:[ `Absolute | `All ] -> string -> string
```